stablehlo/reference/Tensor.cpp

```diff
@@ -50,6 +48,27 @@ int64_t getSizeInBytes(Type type) {
   report_fatal_error(std::move(err));
 }
 
+bool isSupportedUnsignedIntegerType(Type type) {
+  return type.isUnsignedInteger(4) || type.isUnsignedInteger(8) ||
+         type.isUnsignedInteger(16) || type.isUnsignedInteger(32) ||
+         type.isUnsignedInteger(64);
+}
+
+bool isSupportedSignedIntegerType(Type type) {
+  // TODO(#22): StableHLO, as bootstrapped from MHLO, inherits signless
+  // integers which was added in MHLO for legacy reasons. Going forward,
+  // StableHLO will adopt signfull integer semantics with signed and unsigned
+  // integer variants.
+  return type.isSignlessInteger(4) || type.isSignlessInteger(8) ||
+         type.isSignlessInteger(16) || type.isSignlessInteger(32) ||
+         type.isSignlessInteger(64);
+}
+
+bool isSupportedIntegerType(Type type) {
+  return isSupportedUnsignedIntegerType(type) ||
+         isSupportedSignedIntegerType(type);
+}
```

Comment:
Rather than duplicating these functions, I propose we move them to `namespace detail` in `Element.h` (including the `isSupportedFloatType` and `isSupportedComplexType`.

Reply:
Moving the definition to the include file, which gets included in other cpp file, gives link time error  related to duplicate symbol. Am I missing something here?

Reply:
That's right. You'll need to move the declarations to `Element.h` and the definitions to `Element.cpp`. After that, you'll be able to delete these functions from `Tensor.cpp`.